site/src/pages/TerminalPage/TerminalPage.tsx

```diff
@@ -134,13 +132,21 @@ const TerminalPage: FC = () => {
         handleWebLinkRef.current(uri);
       }),
     );
-    setTerminal(terminal);
+
     terminal.open(xtermRef.current);
-    const listener = () => {
-      // This will trigger a resize event on the terminal.
-      fitAddon.fit();
-    };
+
+    // We have to fit twice here. It's unknown why, but the first fit will
+    // overflow slightly in some scenarios. Applying a second fit resolves this.
+    fitAddon.fit();
+    fitAddon.fit();
+
+    // This will trigger a resize event on the terminal.
+    const listener = () => fitAddon.fit();
     window.addEventListener("resize", listener);
+
+    // Terminal is correctly sized and is ready to be used.
+    setTerminal(terminal);
```

Comment:
Is moving the `setTerminal` call just for clarity? Just making sure – because of how/when React flushes state changes, moving the call in this case doesn't actually change anything

Reply:
Yup exactly, I just felt it flowed better but that is probably rather subjective.

Reply:
To be honest part of me also wanted to move it just because I do not necessarily trust what React does behind the scenes hahaha, this way I think the intent is clearer at least even if it is functionally identical.